### PR TITLE
Update GTF usage so that things show and warnings don't

### PR DESCRIPTION
**Pull request by brett-groshong**
### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [x] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [x] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
This change updates the version of ShaderGraph using GTF to match GTF expectations for window display:
* adds a GraphTool (ShaderGraphGraphTool) so that a name is provided through BaseGraphTool and through the window.
* reconnects basic interaction handlers -- this removes basic interaction handler console warning
* moves Window, Stencil, MainToolbar, OnboardingProvider, and GraphTool to the top level of the editor UI directory.
* adds the creation of a MainToolbar (See: ShaderGraphMainToolbar)

---
### Testing status
Functionality will be manually tested as a part of GSG-850.

---

### CODE REVIEW

File: com.unity.shadergraph/Editor/GraphUI/ShaderGraphOnboardingProvider.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 using UnityEngine.GraphToolsFoundation.CommandStateObserver;
 using UnityEngine.UIElements;
 
-namespace UnityEditor.ShaderGraph.GraphUI.GraphElements.Windows
+namespace UnityEditor.ShaderGraph.GraphUI
 {
     public class ShaderGraphOnboardingProvider : OnboardingProvider
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphUI/ShaderGraphOnboardingProvider.cs:9]: Does the onboarding provider still prompt to create an asset? if we could disable that for now it would be nice. But that can be a separate PR.

Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/ShaderGraphOnboardingProvider.cs:9]: It does. I'll get that in a different PR.

Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/ShaderGraphOnboardingProvider.cs:9]: I made a jira task for it https://jira.unity3d.com/browse/GSG-888

**Review by esmelusina — APPROVED** [review]

File: com.unity.shadergraph/Editor/GraphUI/ShaderGraphEditorWindow.cs [hunk 2]
```diff
@@ -1,108 +1,118 @@
...
             // Needed to ensure that graph view takes up full window when overlay canvas is present
             rootVisualElement.style.position = new StyleEnum<Position>(Position.Absolute);
             rootVisualElement.style.width = new StyleLength(new Length(100, LengthUnit.Percent));
             rootVisualElement.style.height = new StyleLength(new Length(100, LengthUnit.Percent));
+        }
 
-            // InitializeSubWindows();
+        protected override BaseGraphTool CreateGraphTool()
+        {
+            return CsoTool.Create<ShaderGraphGraphTool>();
```
[the remark this answers is not in the document]
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/ShaderGraphEditorWindow.cs:39]: We will kick this to FAD meeting.
